### DISCUSSION TITLE
Localize datetimes client side?

### DISCUSSION THREAD

**Discussion by tobymurray** [bigskysoftware/htmx#878] [Q&A]
As far as I'm aware, there are two options for rendering dates/times in the user's timezone:

1. send the timezone to the server so the localized response can be rendered there
2. invoke JavaScript to convert a UTC timestamp to the browser's timezone

Path 1 is pretty straightforward (submitting arbitrary data is easy enough), but with HTMX how can path 2 be accomplished? E.g. an `hx-get` returns `2007-04-05T14:30Z`, but now what?

How are people solving client side localization with HTMX presently?

Might be a little related to: https://github.com/bigskysoftware/htmx/discussions/716

**Comment by nakkaya** [comment]
For number two I use `alphine.js` I just sent the time in `UTC ISO` format then convert it to user local time an example would be,

    <td x-text="moment('{{time}}').local().format('HH:mm');"></td>

**Comment by IRus** [comment]
Client-side localization is one thing, what if I want to display something based on the user's local date.
For example, if the current date for a user is May 4th, I would like to return a different logo on initial load, but I also can't see a way to do it without making additional requests.

**Reply by tobymurray** [reply]
I think you're kind of stuck with providing the driving input (user's timezone?) to the server in the request that yields the logo'd page? Everything else would be conditional client side (depends what you mean by "initial load", I guess) or follow the initial load with a request bearing the driving input.
